src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -77,9 +60,41 @@
     @In
     private NUIManager nuiManager;
 
+    @In
+    private EntityManager entityManager;
+
     private static final Logger logger = LoggerFactory.getLogger(BreedingSystem.class);
     private Random random = new FastRandom();
 
+    /**
+     * This method is called upon when a gooey that is following the player is brought to
+     * the love shack and it steps over a breeding block.
+     *
+     * Receives OnEnterBlockEvent when a gooey entity steps over a breeding block.
+     *
+     * @param event     The OnEnterBlockEvent
+     * @param entity    the gooey entity
+     */
+    @ReceiveEvent(components = {GooeyComponent.class})
+    public void onEnterBlock(OnEnterBlockEvent event, EntityRef entity) {
+        LocationComponent loc = entity.getComponent(LocationComponent.class);
```

Comment:
Since you're never actually using the variable `loc` you can remove this line and rewrite the next line as
`Vetor3f pos = entity.getComponent(LocationComponent.class).getWorldPosition();`

Your choice. If you feel the current one is more readable that's fine too.

Reply:
I feel it is more readable, so I'll stick with it for now 😄